tests/test_attributes.py

```diff
@@ -0,0 +1,211 @@
+        )
+
+
+class AppendAttributesTest(BaseTestCase):
+    def test_single_dict(self):
+        self.assertEqual(
+            append_attributes({"foo": "bar"}),
+            {"foo": "bar"},
+        )
+
+    def test_appends_dicts(self):
+        self.assertEqual(
+            append_attributes({"class": "foo"}, {"id": "bar"}, {"class": "baz"}),
+            {"class": "foo baz", "id": "bar"},
+        )
+
+
+class NormalizeClassTest(BaseTestCase):
+    def test_str(self):
+        self.assertEqual(
+            normalize_class("foo"),
+            "foo",
+        )
+
+    def test_list(self):
+        self.assertEqual(
+            normalize_class(["foo", "bar"]),
+            "foo bar",
+        )
+
+    def test_nested_list(self):
+        self.assertEqual(
+            normalize_class(["foo", ["bar", "baz"]]),
+            "foo bar baz",
+        )
+
+    def test_tuple(self):
+        self.assertEqual(
+            normalize_class(("foo", "bar")),
+            "foo bar",
+        )
+
+    def test_dict(self):
+        self.assertEqual(
+            normalize_class({"foo": True, "bar": False, "baz": None}),
+            "foo",
+        )
+
+    def test_combined(self):
+        self.assertEqual(
+            normalize_class(
+                [
+                    "class1",
+                    ["class2", "class3"],
+                    {
+                        "class4": True,
+                        "class5": False,
+                        "class6": "foo",
+                    },
+                ]
+            ),
+            "class1 class2 class3 class4 class6",
+        )
+
+
+class AttrsInTemplateTests(BaseTestCase):
+    def test_attrs_accessible_in_template(self):
+        @component.register("test")
+        class AttrsComponent(component.Component):
+            template: types.django_html = """
+                {% load component_tags %}
+                <div>
+                    {{ component_vars.attrs|safe }}
+                </div>
+            """
+
+        template_str: types.django_html = """
+            {% load component_tags %}
+            {% component "test" attrs:@click.stop="dispatch('click_event')" attrs:x-data="{hello: 'world'}" attrs:class=class_var %}
+            {% endcomponent %}
+        """
+        template = Template(template_str)
+        rendered = template.render(Context({"class_var": "padding-top-8"}))
+        self.assertHTMLEqual(
+            rendered,
+            """
+            <div>
+                {'@click.stop': "dispatch('click_event')",
+                'x-data': "{hello: 'world'}",
+                'class': 'padding-top-8'}
+            </div>
+            """,
+        )
+
+    def test_attrs_tag(self):
+        @component.register("test")
+        class AttrsComponent(component.Component):
+            template: types.django_html = """
+                {% load component_tags %}
+                <div {% html_attrs component_vars.attrs class+="added_class" data-id=123 %}>
```

Comment:
Also note here that I used `component_vars.attrs` to access the attributes. It's a bit wordy, it would've been awesome if it could be something simple like `$attrs`, but 1) dollar sign is not a valid identifier character, and 2) succint names can be achieved later. Now it's imo important that things work.